src/test/java/teammates/test/cases/BaseTestCaseWithDatastoreAccess.java

```diff
@@ -0,0 +1,303 @@
+package teammates.test.cases;
+
+import java.util.Map;
+
+import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.datatransfer.CommentAttributes;
+import teammates.common.datatransfer.CourseAttributes;
+import teammates.common.datatransfer.DataBundle;
+import teammates.common.datatransfer.EntityAttributes;
+import teammates.common.datatransfer.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.FeedbackResponseAttributes;
+import teammates.common.datatransfer.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.datatransfer.StudentProfileAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.JsonUtils;
+import teammates.common.util.StringHelper;
+import teammates.common.util.ThreadHelper;
+import teammates.test.driver.BackDoor;
+
+/**
+ * Base class for all test cases which are allowed to access the Datastore via {@link BackDoor}.
+ */
+@SuppressWarnings("PMD.TestClassWithoutTestCases") // false positive by PMD
+public class BaseTestCaseWithDatastoreAccess extends BaseTestCase {
+    
+    private static final int VERIFICATION_RETRY_COUNT = 5;
+    private static final int VERIFICATION_RETRY_DELAY_IN_MS = 1000;
+    private static final int OPERATION_RETRY_COUNT = 5;
+    private static final int OPERATION_RETRY_DELAY_IN_MS = 1000;
+    
+    protected void verifyPresentInDatastore(DataBundle data) {
+        Map<String, AccountAttributes> accounts = data.accounts;
+        for (AccountAttributes account : accounts.values()) {
+            verifyPresentInDatastore(account);
+        }
+        
+        Map<String, InstructorAttributes> instructors = data.instructors;
+        for (InstructorAttributes instructor : instructors.values()) {
+            verifyPresentInDatastore(instructor);
+        }
+        
+        Map<String, CourseAttributes> courses = data.courses;
+        for (CourseAttributes course : courses.values()) {
+            verifyPresentInDatastore(course);
+        }
+        
+        Map<String, StudentAttributes> students = data.students;
+        for (StudentAttributes student : students.values()) {
+            verifyPresentInDatastore(student);
+        }
+    }
+    
+    private EntityAttributes getEntity(EntityAttributes expected) {
+        if (expected instanceof AccountAttributes) {
```

Comment:
It doesn't help much for a single usage and I agree that it is too much work if we won't ever be using this more than once, if we are going to do this more than once I will suggest we take the route of pushing it to one method having the burden rather than at multiple sites so let's keep that in mind if we ever get multiples sites of such occurances!